File: CourseFlow/Models/Course.cs
namespace CourseFlow.Models;

public class Course : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    
    public string Author { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string Thumbnail { get; set; } = string.Empty;

    public int Price { get; set; } = 0;

    public DateTime DateCreated { get; set; } = DateTime.Now;

    public bool IsPublic { get; set; } = false;

    public ICollection<Lesson> Lessons { get; set; }

    public ICollection<CourseCategory> CourseCategories { get; set; }
    
    public ICollection<AuthorCourses> AuthorCourses { get; set; }
    
    public ICollection<UserTakesCourse> UserTakesCourse { get; set; }
    
}

File: Frontend/src/interfaces/Course.ts
import { Lesson } from './Lesson';
import { User } from './User';

export interface Course {
  id: string;
  title: string;
  description: string;
  thumbnail: string;
  price: number;
  categories: string[];
  dateCreated: string;
  isPublic: boolean;
  author: string;
  lessons: Lesson[];
  courseCategories?: CourseCategories[]
}


export interface CourseCategories{
  name: string;
}